### ISSUE TITLE
Double Loading Psych - Conflict with Ruby Bundled Version

### ISSUE THREAD

**Issue by tadman** [ruby/psych#66]
Lately I've had nothing but trouble with Psych conflicting with the version that's built into Ruby. Using 1.9.3 and Psych 1.3.3 leads to the all too familiar " warning: already initialized constant" stream of errors that results when attempting to redefine what is already defined.

There was talk of introducing a fail-safe where Psych wouldn't load if already loaded, which you'd think would be as simple as adding something like this to the gem's main library file:

```
if (defined?(Psych))
  # Opportunity to warn or complain about already being loaded here...
end
```

Unlike most other gems where double-loading is almost always the fault of the developer, Psych is in the relatively unique position of being bundled with Ruby and being distributed as a commonly used gem. It's further complicated by the fact that it has C extensions that need to be loaded as well, and as far as I can tell, these extensions cannot be swapped out once loaded.

In short, allowing Psych to load twice is a bug in the library. It should, at the very least, warn when loaded twice. Ideally it should try and identify what caused it to be loaded prematurely, though this could be tricky to discern.

I'd also like to propose that the version of Psych shipped with Ruby be called something other than Psych so there's no risk of conflict in the future. It needs a different namespace so that it can co-exist with the gem version. This would side-step virtually all o the problems I've experienced and would eliminate a lot of confusion encountered by others if the issues filed here and questions asked in other places are any example.

Whatever YAML library is the default in Ruby should be given a name reflecting this status as "official".
